### PR TITLE
fixing bug in angle difference constraint

### PR DESCRIPTION
**Pull request by michaelbynum**

### CODE REVIEW

File: egret/model_library/transmission/branch.py [hunk 1]
```diff
@@ -944,11 +949,21 @@ def declare_ineq_angle_diff_branch_lbub_c_s(model, index_set, branches):
...
+            m.ineq_angle_diff_branch_lb[branch_name] = (math.tan(math.radians(branches[branch_name]['angle_diff_min'])) *
+                                                        m.c[(from_bus, to_bus)] <= m.s[(from_bus, to_bus)])
+        if branches[branch_name]['angle_diff_max'] < 90:
+            if branches[branch_name]['angle_diff_min'] > 89:
+                msg = 'angle difference limits larger than 89 will introduce large coefficients'
+                logger.warning(msg)
+                warnings.warn(msg)
+            m.ineq_angle_diff_branch_ub[branch_name] = (m.s[(from_bus, to_bus)] <=
+                                                        math.tan(math.radians(branches[branch_name]['angle_diff_max'])) *
+                                                        m.c[(from_bus, to_bus)])
```
Comment (carldlaird) [inline egret/model_library/transmission/branch.py:966]: If the angle min is less than -90 deg, we are skipping the constraint? Is that correct? (same for max and 90 deg)

Comment (michaelbynum) [reply egret/model_library/transmission/branch.py:966]: Correct

File: egret/model_library/transmission/branch.py [hunk 2]
```diff
@@ -944,11 +949,21 @@ def declare_ineq_angle_diff_branch_lbub_c_s(model, index_set, branches):
...
         to_bus = branches[branch_name]['to_bus']
 
-        m.ineq_angle_diff_branch_lb[branch_name] = (math.tan(math.radians(branches[branch_name]['angle_diff_min'])) *
-                                                    m.c[(from_bus, to_bus)] <= m.s[(from_bus, to_bus)])
-        m.ineq_angle_diff_branch_ub[branch_name] = (m.s[(from_bus, to_bus)] <=
-                                                    math.tan(math.radians(branches[branch_name]['angle_diff_max'])) *
-                                                    m.c[(from_bus, to_bus)])
+        if branches[branch_name]['angle_diff_min'] > -90:
+            if branches[branch_name]['angle_diff_min'] < -89:
+                msg = 'angle difference limits larger than 89 will introduce large coefficients'
```
Comment (jeanpaulwatson) [inline egret/model_library/transmission/branch.py:954]: I would suggest adding the offending branch name in the log message.

File: egret/model_library/transmission/branch.py [hunk 3]
```diff
@@ -944,11 +949,21 @@ def declare_ineq_angle_diff_branch_lbub_c_s(model, index_set, branches):
...
+                logger.warning(msg)
+                warnings.warn(msg)
+            m.ineq_angle_diff_branch_lb[branch_name] = (math.tan(math.radians(branches[branch_name]['angle_diff_min'])) *
+                                                        m.c[(from_bus, to_bus)] <= m.s[(from_bus, to_bus)])
+        if branches[branch_name]['angle_diff_max'] < 90:
+            if branches[branch_name]['angle_diff_min'] > 89:
+                msg = 'angle difference limits larger than 89 will introduce large coefficients'
```
Comment (jeanpaulwatson) [inline egret/model_library/transmission/branch.py:961]: Ditto here, regarding adding the offending branch name in the message.

**Review by jeanpaulwatson — APPROVED** [review]
Minor nit regarding enhancing the output of the warning log messages - beyond that, looks good.

**Pull request merged**